akka-typed/src/main/scala/akka/typed/Behavior.scala

```diff
@@ -179,12 +175,8 @@ object Behavior {
   /**
    * Validate the given behavior as initial, initialize it if it is deferred.
    */
-  def preStart[T](behavior: Behavior[T], ctx: ActorContext[T]): Behavior[T] = {
-    validateAsInitial(behavior) match {
-      case d: DeferredBehavior[T] @unchecked ⇒ validateAsInitial(undefer(d, ctx))
-      case x                                 ⇒ x
-    }
-  }
+  def preStart[T](behavior: Behavior[T], ctx: ActorContext[T]): Behavior[T] =
```

Comment:
I find the name `preStart` a bit confusing nowadays. Could you try to remove it and directly call `validateAsInitial` instead, since that is the purpose. Then `validateAsInitial` should probably perform the `undefer` as the first thing.